annotation-utils/src/main/java/edu/isi/vista/annotationutils/ExtractAnnotationStats.kt

```diff
@@ -161,17 +164,25 @@ class ExtractAnnotationStats {
                             // We determine the "significant spans" because
                             // these indicate if the sentence is negative, and it's faster than
                             // running through each span to find the annotated sentences.
-                            var significantDocumentSpans = listOf<JsonNode>()
-                            if (documentRelations == null) {
-                                // Try CTEventSpan
-                                val ctEventSpan = jsonTree["_views"]["_InitialView"]["CTEventSpan"]
-                                if (ctEventSpan != null && corpus != "ACE"){
-                                    significantDocumentSpans = ctEventSpan.toList()
+                            var significantDocumentSpans = mutableListOf<JsonNode>()
+                            if (documentCTEventSpan != null) {
+                                if (corpus != "ACE") {
+                                    if (documentRelations != null) {
+                                        significantDocumentSpans = getPrimarySpans(documentRelations, documentSpans)
+                                    }
+                                    // Add any argument-less spans from CTEventSpan.
+                                    // An example of a CTEventSpan value is
+                                    // "[1000, 1001, {"sofa": 1, "begin": 200, "end": 205, "negative_example": true}]"
+                                    // Integers represent spans in relations; we are
+                                    // only interested in collecting the spans here that aren't.
+                                    for (item in documentCTEventSpan) {
+                                        if (!item.isInt) {
```

Comment:
Can the third item in `documentCTEventSpan` be used, instead of filtering on the type? Or will there be multiple spans in `documentCTEventSpan`?

Reply:
An argument-less span could be at any index, and there could be more than one.

Reply:
This might be a better example:
`[12135,12140,12145,{"sofa":1,"begin":310,"end":318,"negative_example":true},12130]`

Reply:
I understand, and this also handles the case if there is ever more than one argument-less span. I brought it up because I was hoping there was a more concrete way to check the type of that bit of JSON, but if you can guarantee that `documentCTEventSpan` is only ever integers and that dict, I'm ok with this.